arch/riscv/core/thread.c

```diff
@@ -190,3 +190,39 @@ FUNC_NORETURN void arch_user_mode_enter(k_thread_entry_t user_entry,
 }
 
 #endif /* CONFIG_USERSPACE */
+
+#if !defined(CONFIG_MULTITHREADING)
+
+K_KERNEL_STACK_ARRAY_DECLARE(z_interrupt_stacks, CONFIG_MP_MAX_NUM_CPUS, CONFIG_ISR_STACK_SIZE);
+K_THREAD_STACK_DECLARE(z_main_stack, CONFIG_MAIN_STACK_SIZE);
+
+FUNC_NORETURN void z_riscv_switch_to_main_no_multithreading(k_thread_entry_t main_entry,
+							    void *p1, void *p2, void *p3)
+{
+	void *main_stack;
+
+	_kernel.cpus[0].id = 0;
+	_kernel.cpus[0].irq_stack = (Z_KERNEL_STACK_BUFFER(z_interrupt_stacks[0]) +
+				     K_KERNEL_STACK_SIZEOF(z_interrupt_stacks[0]));
+
+	main_stack = (Z_THREAD_STACK_BUFFER(z_main_stack) +
+		      K_THREAD_STACK_SIZEOF(z_main_stack));
+
+	register void *a0 __asm__("a0") = p1;
+	register void *a1 __asm__("a1") = p2;
+	register void *a2 __asm__("a2") = p3;
```

Comment:
Those are never used, you can leave those out.

Reply:
Fixed it.